packages/xrpl/src/models/transactions/depositPreauth.ts

```diff
@@ -63,5 +60,36 @@ export function validateDepositPreauth(tx: Record<string, unknown>): void {
         "DepositPreauth: Account can't unauthorize its own address",
       )
     }
+  } else if (tx.AuthorizeCredentials !== undefined) {
+    validateCredentialsList(
+      tx.AuthorizeCredentials,
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- confirmed in base transaction check
+      tx.TransactionType as string,
+      false,
+    )
+  } else if (tx.UnauthorizeCredentials !== undefined) {
+    validateCredentialsList(
+      tx.UnauthorizeCredentials,
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- confirmed in base transaction check
+      tx.TransactionType as string,
+      false,
+    )
+  }
+}
+
+// Boolean logic to ensure exactly one of 4 inputs was provided
+function validateSingleAuthorizationFieldProvided(
+  tx: Record<string, unknown>,
+): void {
+  const countProvided =
+    Number(tx.Authorize !== undefined) +
+    Number(tx.Unauthorize !== undefined) +
+    Number(tx.AuthorizeCredentials !== undefined) +
+    Number(tx.UnauthorizeCredentials !== undefined)
```

Comment:
```suggestion
const fields = ['Authorize', 'Unauthorize', 'AuthorizeCredentials', 'UnauthorizeCredentials']
const countProvided =(fields.filter(key => tx[key] !== undefined)).length;
```
Something like this would be cleaner